Mathlib/GroupTheory/Exponent.lean

```diff
@@ -520,6 +520,22 @@ theorem Subgroup.exponent_top : Monoid.exponent (⊤ : Subgroup G) = Monoid.expo
 theorem Subgroup.pow_exponent_eq_one {H : Subgroup G} {g : G} (g_in_H : g ∈ H) :
     g ^ Monoid.exponent H = 1 := exponent_toSubmonoid H ▸ Submonoid.pow_exponent_eq_one g_in_H
 
+@[to_additive]
+theorem Group.exponent_dvd_iff_forall_zpow_eq_one :
+    (Monoid.exponent G : ℤ) ∣ n ↔ ∀ g : G, g ^ n = 1 := by
+  simp_rw [Int.natCast_dvd, Monoid.exponent_dvd_iff_forall_pow_eq_one, pow_natAbs_eq_one]
+
+@[to_additive]
+theorem zpow_eq_one_of_exponent_dvd (h : (Monoid.exponent G : ℤ) ∣ n) : g ^ n = 1 :=
+  orderOf_dvd_iff_zpow_eq_one.mp <|
+    dvd_trans (Int.natCast_dvd_natCast.mpr <| Monoid.order_dvd_exponent g) h
```

Comment:
```suggestion
  Group.exponent_dvd_iff_forall_zpow_eq_one.mp h g
```
Why not use the new lemma?

Reply:
Because apparently I got distracted and didn't finish what I started before pushing - I just meant to have the iffs.